src/error/errorHandler.ts

```diff
@@ -0,0 +1,19 @@
+import logger from '../../logger';
+
+export const handleError = (error: any, debug: boolean = false): void => {
+
+  if (error.code === 'ENOENT') {
+    logger.error(`File or directory not found.`);
+  } else if (error.code === 'EACCES') {
+    logger.error('Access denied. Please check the file permissions.');
+  } else if (error.code) {
+    logger.error(`System Error - Code: ${error.code}, Message: ${error.message}, Path: ${error.path}`);
+  } else {
+    logger.error('An unexpected error occurred:', error.message || error);
+  }
+
+  if (debug) {
+    logger.debug('Debug mode is enabled, throwing error for stack trace.');
+    throw error;
```

Comment:
Good start, but logging the error stack ( `error.stack` ) in debug mode would provide more context.